app/models/user.rb

```diff
@@ -61,8 +61,12 @@ def pciu_alternate_phone
     pciu&.get_alternate_phone&.to_s
   end
 
-  def first_name
-    identity.first_name || (mhv_icn.present? ? mpi&.profile&.given_names&.first : nil)
+  def first_name(args = { mpi: false })
+    if args[:mpi] == true
```

Comment:
I might even suggest that we don't combine these into singular functions. I would consider having two functions, `first_name`, and `first_name_mpi`. Once we're 100% SSOe we can just find/replace all instances of `first_name_mpi` to `first_name`.